manager/dispatcher/assignments.go

```diff
@@ -46,9 +35,58 @@ func (a *assignmentSet) addTaskDependencies(readTx store.ReadTx, t *api.Task) {
 	if container != nil {
 		secrets = container.Secrets
 	}
+
+	for _, resourceRef := range t.Spec.ResourceReferences {
+		a.log.WithField("resource.id", resourceRef.ResourceID)
+		mapKey := typeAndID{objType: resourceRef.ResourceType, id: resourceRef.ResourceID}
+		if len(a.tasksUsingDependency[mapKey]) == 0 {
+			a.tasksUsingDependency[mapKey] = make(map[string]struct{})
+			switch resourceRef.ResourceType {
+			case api.ResourceType_SECRET:
+				a.log.WithField("resource.type", "secret")
+				secret := store.GetSecret(readTx, resourceRef.ResourceID)
+				if secret == nil {
+					a.log.WithFields(logrus.Fields{
+						"secret.id": resourceRef.ResourceID,
+					}).Debug("secret not found by resource reference ID")
+					continue
+				}
+				a.changes[mapKey] = &api.AssignmentChange{
+					Assignment: &api.Assignment{
+						Item: &api.Assignment_Secret{
+							Secret: secret,
+						},
+					},
+					Action: api.AssignmentChange_AssignmentActionUpdate,
+				}
```

Comment:
Let's break these cases out into separate methods so we can share the code with the legacy implementation that uses `container.Secrets` and `container.Configs`, below.